src/xpath.rs

```diff
@@ -133,7 +134,12 @@ impl Object {
       if ptr.is_null() {
         panic!("rust-libxml: xpath: found null pointer result set");
       }
-      vec.push(Node { node_ptr: ptr }); //node_is_inserted : true
+
+      /* TODO: break next two lines into a method on _Docuemnt */
+      let node = Node::wrap(ptr, self.document.clone());
+      self.document.borrow_mut().insert_node(ptr, node.clone());
```

Comment:
Right, on this TODO note, I am wondering what is the most sensible API for public methods such as `insert_node`.

The main question I have is whether the author still wants to keep a handle on the node after they insert it in a document, and the answer is probably "yes". That justifies using a clone or a reference. But it may have clearer meaning if we use a mutable reference to `&mut node` instead of cloning, to relay to the author that the node is still the same object in memory, but has changed an attribute.

Revisiting most public methods with such questions and trying to keep the API intuitive is my next step here.

Reply:
And of course many of these issues are already present in the master branch, this PR provides the tools to resolve them so it seems like an opportune moment to do the upgrades.